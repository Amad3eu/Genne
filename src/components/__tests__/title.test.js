import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Title from '../title';

describe('Title', () => {
  const mockEvent = { target: { value: 'This is a mock event' } };
  const props = {
    prefix: {
      title: 'test_title',
      currentWork: 'test_currentwork',
    },
    data: { title: 'test_data' },
    link: { currentWork: 'test_currentwork' },
    handlePrefixChange: jest.fn().mockReturnValue({}),
    handleLinkChange: jest.fn().mockReturnValue({}),
    handleDataChange: jest.fn().mockReturnValue({}),
  };

  it('renders title component correctly', () => {
    const component = shallow(<Title {...props} />);
    component.find('input').at(0).simulate('change', mockEvent);
    component.find('input').at(1).simulate('change', mockEvent);
    expect(toJson(component)).toMatchSnapshot();
  });
});
