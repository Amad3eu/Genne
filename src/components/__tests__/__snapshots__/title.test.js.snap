// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders title component correctly 1`] = `
<div
  className="flex justify-center items-start flex-col w-full px-2 sm:px-6 mb-10"
>
  <div
    className="text-xl sm:text-2xl font-bold font-title mt-2 mb-2"
  >
    Title
  </div>
  <div
    className="flex justify-start items-center w-full text-regular text-xs sm:text-lg"
  >
    <input
      className="outline-none w-24 sm:w-40 mr-10 border-t-0 border-l-0 border-r-0 border solid border-gray-900 py-1 px-2 focus:border-blue-700 prefix"
      id="title-prefix"
      onChange={[Function]}
      value="test_title"
    />
    <input
      className="outline-none placeholder-gray-700 w-1/2 sm:w-1/3 border-t-0 border-l-0 border-r-0 border solid border-gray-900 py-1 px-2 focus:border-blue-700"
      id="title-name"
      onChange={[Function]}
      placeholder="name"
      value="test_data"
    />
  </div>
</div>
`;
